building/glass/branches-schema.yaml

```diff
@@ -2,42 +2,37 @@ $schema: "http://json-schema.org/draft-07/schema#"
 title: BranchesConfig
 type: object
 properties:
-  upload-targets:
-    type: array
-    items:
-      type: object
-      oneOf:
-        - properties:
-            name:
-              type: string
-            method:
-              enum: ["google-cloud-storage"]
-          required: ["name", "method"]
-          additionalProperties: false
-        - properties:
-            name:
-              type: string
-            method:
-              enum: ["rsync"]
-            user:
-              type: string
-            dir:
-              type: string
-          required: ["name", "method", "user", "dir"]
-          additionalProperties: false
   branches:
     type: array
     items:
       type: object
       properties:
         name:
           type: string
-          pattern: "^[0-9a-zA-Z_.-][0-9a-zA-Z_./-]+[0-9a-zA-Z_.-]$"
+          pattern: "[0-9a-zA-Z_-]+"
+        download:
+          type: string
+          pattern: "^no_download$|^https?://[0-9a-zA-Z_.-]+(:[0-9]+)?(/[0-9a-zA-Z_./-]*[0-9a-zA-Z_.-])?$"
         signing-key:
           type: string
           pattern: "^[a-fA-F0-9]+$"
-        upload-target:
-          type: string
+        upload:
+          type: object
+          oneOf:
+            - properties:
+                method:
+                  enum: ["google-cloud-storage"]
+                gcs-target:
+                  type: string
```

Comment:
Might be good to validate the gcs-target and rsync-target fields.

Reply:
that is a bit difficult since rsync targets can have many different forms.  I think it's not worth it, especially since if it's wrong then the rsync command will fail quickly enough anyway